### PR TITLE
Date constructor definition (fixed tests)

### PR DESCRIPTION
**Pull request by jjergus**
Pull request #537, now with fixed tests (changed line numbers in core.js).

Partially solves #188

Includes updated test case.

It looks like overloaded constructors are not supported, so this doesn't detect all invalid constructor calls (see examples in the test case).

### CODE REVIEW

**Comment by knsn1994** [comment]
Nice.

File: lib/core.js [hunk 1]
```diff
@@ -207,6 +207,11 @@ declare class RegExp {
 }
 
 declare class Date {
+    // new Date();
```
Comment (bhosmer) [inline lib/core.js]: Hi Jan - if you could add a comment here noting that we want to use constructor overloads once they become available, that would be great.

**Comment by bhosmer** [comment]
One comment request, but looks good - thanks a bunch! Will merge as soon as you update. :)

**Comment by jjergus** [comment]
Done.

**Comment by bhosmer** [comment]
Sweet! Will merge once Travis build is done. Thanks again!

**Comment by bhosmer** [comment]
Dunno what's going on with Travis... merging

**Pull request merged**